ruby/red-arrow/test/raw-records/test-dictionary-array.rb

```diff
@@ -0,0 +1,332 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+module RawRecordsDictionaryArrayTests
+  def build_schema(dictionary)
+    Arrow::Schema.new({
+      column: {
+        type: :dictionary,
+        index_data_type: :int32,
+        value_data_type: dictionary.value_data_type,
+        ordered: false,
+      },
+    })
+  end
+
+  def build_record_batch(dictionary, records)
```

Comment:
Umm. This `dictionary` isn't a dictionary.

For example, `Arrow::NullArray.new(records.flatten)` in `build(Arrow::NullArray.new(records.flatten), records)` isn't a dictionary.

How about something like the following?

```ruby
def build_record_batch(array)
  dictionary = array.dictionary_encode
  schema = Arrow::Schema.new(column: dictionary.value_data_type)
  Arrow::RecordBatch.new(schema, array.length, [dictionary])
end
```

```ruby
target = build(Arrow::NullArray.new(records.collect(&:first)))
```